timemachine/fe/free_energy.py

```diff
@@ -1103,7 +1103,7 @@ def compute_intermediate_result(lambdas: Sequence[float]) -> PairBarResult:
         result = compute_intermediate_result(lambdas)
         results.append(result)
     else:
-        if min_overlap is not None:
+        if min_overlap is not None and np.min(result.overlaps) < min_overlap:
```

Comment:
nit: use native Python `min`, since `result.overlaps` is a list
```suggestion
        if min_overlap is not None and min(result.overlaps) < min_overlap:
```

Reply:
Hmm, on second thought, maybe it is better to use the numpy version here because the behavior with NaNs is more consistent:

```
>>> min([1.0, np.nan])
1.0
>>> np.min([1.0, np.nan])
np.float64(nan)
```

Non-blocking, OK either way.